lib/controllers/include/Current.h

```diff
@@ -1,13 +1,20 @@
 #pragma once
 
-#include "BasicControl.h"
-
+#include "ActuatorController.h"
 namespace Controller {
 
-class Current final : public BasicControl {
+class Current final : public ActuatorController {
  public:
-  using BasicControl::BasicControl;
+  Current(Actuator::Actuator &actuator, Encoder::Encoder &encoder,
+          const std::optional<std::reference_wrapper<Sensor::CurrentSensor> const> &currentSensor, PID::PID &pid,
+          float maxDegPerSec, float maxCurrent, PinName lowerLimit, PinName upperLimit,
+          bool ignoreDegPerSecChecks = false, bool ignoreCurrentChecks = true, bool ignoreLimitSwitchChecks = false);
```

Comment:
Not something to be addressed for this pr, but I feel like I need to make this comment:

With a lot of our objects getting more and more complex, I really think we should consider going back to using config structs. It makes the code more readable(passing a million things to a constructor is messy af in c++ vs. using designated initializers in c99 and c++20). Its also less prone to errors(accidentally swapping order of two same-type args vs. designated initializers naming each member).